mlir/lib/Catalyst/Transforms/ApplyTransformSequencePass.cpp

```diff
@@ -19,21 +19,129 @@
 
 #include <cassert>
 
-#include "llvm/Support/Debug.h"
-
+#include "mlir/Dialect/Transform/IR/TransformOps.h"
 #include "mlir/Dialect/Transform/Interfaces/TransformInterfaces.h"
+#include "mlir/Dialect/Transform/Transforms/TransformInterpreterUtils.h"
 #include "mlir/IR/BuiltinOps.h"
+#include "mlir/Pass/AnalysisManager.h"
 #include "mlir/Pass/Pass.h"
-
-#include "Catalyst/IR/CatalystDialect.h"
-#include "QEC/IR/QECDialect.h"
+#include "mlir/Pass/PassInstrumentation.h"
+#include "mlir/Pass/PassManager.h"
 
 using namespace llvm;
 using namespace mlir;
-using namespace catalyst;
 
 namespace catalyst {
 
+/// Generate a meaningful name for a transform operation for pass instrumentation
+static std::string getTransformOpName(transform::TransformOpInterface transformOp)
+{
+    std::string baseName = transformOp->getName().getStringRef().str();
+
+    if (auto applyPassOp = dyn_cast<transform::ApplyRegisteredPassOp>(transformOp.getOperation())) {
+        if (auto passName = applyPassOp.getPassName(); !passName.empty()) {
```

Comment:
```suggestion
        if (auto passName = applyPassOp.getPassName() && !passName.empty()) {
```

as per that discussion from Jason a while ago